prow/cmd/deck/static/common/rerun.ts

```diff
@@ -0,0 +1,151 @@
+import {alertToast, copyToClipboardWithToast, icon, setToast} from "./common";
+import {relativeURL} from "./urls";
+
+export function createRerunProwJobIcon(modal: HTMLElement, rerunElement: HTMLElement, prowjob: string, rerunCreatesJob: boolean, csrfToken: string): HTMLElement {
+  const LATEST = 'latest';
+  const ORIGINAL = 'original';
+  const inrepoconfigURL = 'https://github.com/kubernetes/test-infra/blob/master/prow/inrepoconfig.md';
+  const latestURL = `${location.protocol}//${location.host}/rerun?mode=latest&prowjob=${prowjob}`;
+  const originalURL = `${location.protocol}//${location.host}/rerun?mode=original&prowjob=${prowjob}`;
+  let commandURL = latestURL;
+  const i = icon.create("refresh", "Show instructions for rerunning this job");
+
+  window.onkeydown = (event: any) => {
+    if ( event.key === "Escape" ) {
+      modal.style.display = "none";
+      // Resets modal content. If removed elements will be concatenated, causing duplicates.
+      rerunElement.innerHTML = '';
+    }
+  };
+  window.onclick = (event: any) => {
+    if (event.target === modal) {
+      modal.style.display = "none";
+      // Resets modal content. If removed elements will be concatenated, causing duplicates.
+      rerunElement.innerHTML = '';
+    }
+  };
+  
+  const getCommandDescription = (mode: string): string => {
+    return `The command is for the ${mode} configuration. Admin permissions are required to rerun via kubectl.`
+  }
+  const getCommand = (url: string):string => {
+    return `kubectl create -f "${url}"`
+  };
+
+  // we actually want to know whether the "access-token-session" cookie exists, but we can't always
+  // access it from the frontend. "github_login" should be set whenever "access-token-session" is
+  i.onclick = () => {
+    modal.style.display = "block";
+
+    rerunElement.innerHTML = `
+      <h2 class="rerunModal-title">Rerun ProwJob</h2>
+      <p class="rerunModal-description">
```

Comment:
As I briefly mentioned earlier (in person), it's unfortunate that our styles are keyed off _paragraphs-with-classes_. I'd rather see a parent div with the `-description` class to handle the stylistic needs and the structural use of paragraphs to only handle the line-breaking. But I don't want to block you on that. Maybe next-next time. 😉